doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,941 @@
+| NPU | Network Processing Unit |
+| DPU | Data Processing Unit |
+| PDK | Platform Development Kit |
+| SAI | Switch Abstraction Interface |
+| GPIO | General Purpose Input Output |
+| PSU | Power Supply Unit |
+| I2C | Inter-integrated Circuit communication protocol |
+| SysFS | Virtual File System provided by the Linux Kernel |
+| CP | Control Plane |
+| DP | Data Plane |
+
+## 1. Introduction
+SmartSwitch offloads the Packet Processors (NPUs) and the host CPUs, freeing up resources for application performance, thereby performing layer four to layer seven functions in a cost effective and space saving way. 
+
+The specialized DPUs when built into a regular switch, can provide such a capability, which is being referred as SmartSwitch.
+Platform monitor PMON in SONiC is a container responsible for chassis management functions to ensure proper operation of the devices and peripherals in the chassis there by ensuring the proper operation of the product. 
+
+The typical lifecycle of a product involves the following stages.
+
+<p align="center"><img src="./images/lifecycle.svg"></p>
+
+The following sub-tasks are performed under each stage
+* Onboarding
+    * Boot, Shutdown, Power Cycle
+    * Rest, PCIe-Reset
+* Monitoring
+    * Device State (dpu_state)
+    * Sensors, PSUs, Cooling Devices, Thermal management
+    * Show CLIs
+* Detection and Debugging
+    * DPU Health
+    * Alarms, Syslog
+    * Console
+* RMA
+    * Inventory
+
+The purpose of this document is to provide a framework to share the state, health, alarms of the DPUs, manage the DPUs by providing support to monitor, gracefully shutdown, restart them and the associated peripherals such as thermal sensors, cooling devices, LEDs, etc.
+
+The picture below highlights the PMON vertical and its association with other logics within the SONiC architecture.
+
+<p align="center"><img src="./images/pmon-vertical.svg"></p>
+
+## 2.	Requirements and Assumptions
+
+### 2.1.    Onboarding
+* The SmartSwitch host PMON should be able to Power Cycle, Shutdown, Reset, and reset the PCIe link per DPU or the entire system. The DPU_MODULE will behave like the LINE_CARD_MODULE of a modular chassis with respect to these functions.
+
+#### SmartSwitch PowerUp sequence:
+* When the smartswitch device is booted, the host will boot first and leave the DPUs in powered up state by default.
+* The DPUs can be powered down by configuring the admin_status as shown.
+* The corresponding switch stateDB table is also shown
+
+#### config_db.json
+CHASSIS_MODULE table holds the list and configuration of DPU modules in a smartswitch chassis. It allows user to administratively bring down a DPU
+```
+{
+    "CHASSIS_MODULE": {
+        "DPU0": {
+            "admin_status": "up"
+        },
+        "DPU1": {
+            "admin_status": "up"
+        }
+    }
+}
+```
+#### switch stateDB
+```
+    Key: "CHASSIS_MODULE_TABLE|DPU0"
+    "CHASSIS_MODULE_TABLE|DPU0":{
+        "value": {
+            "admin_status":"up",
```

Comment:
Why do we need an admin state in State DB? I don't see it used in any of the flows.

Reply:
That's a typo.  Good catch!  It should be ConfigDB and the table will be as shown below
  "CHASSIS_MODULE|DPU4": {
    "expireat": 1693437132.8384774,
    "ttl": -0.001,
    "type": "hash",
    "value": {
      "admin_status": "down"
    }
  },